huggle/login.cpp

```diff
@@ -1269,42 +1269,71 @@ bool Login::ProcessOutput(WikiSite *site)
     ApiQuery *query = this->LoginQueries[site];
     // Check what the result was
     ApiQueryResult *result = query->GetApiQueryResult();
-    ApiQueryResultNode *ln = result->GetNode("login");
-    QString result_code = ln->GetAttribute("result");
-    QString reason = ln->GetAttribute("reason");
-    if (result_code.isEmpty())
+    ApiQueryResultNode *ln = result->GetNode("clientlogin");
+    QString status = ln->GetAttribute("status");
+    if (status.isEmpty())
     {
         this->DisplayError(_l("api.php-invalid-response"));
         return false;
     }
-    if (result_code == "Success")
+
+    if (status == "PASS")
         return true;
-    if (result_code == "EmptyPass")
-    {
-        this->DisplayError(_l("login-password-empty"));
+    if (status == "UI") {
+        // Need a user interaction like captacha or 2FA
+        //QString v_id = ln->ChildNodes.at(0)->GetAttribute("id", "unknown");
+        //if (v_id == "TOTPAuthenticationRequest"){
```

Comment:
I can't get back the `TOTPAuthenticationRequest` value... [Here is](https://phabricator.wikimedia.org/P6304) what contain the answer.